requests/exceptions.py

```diff
@@ -37,6 +37,10 @@ class ProxyError(ConnectionError):
     """A proxy error occurred."""
 
 
+class InvalidProxyURL(ProxyError, ValueError):
```

Comment:
Why inherit from `ValueError` here? Is this so folks can catch `ValueError` across all versions of requests (before and after this fix)? If so, I think it might be better to just make this inherit from `ProxyError` and allow folks to do `except (InvalidProxyURL, ValueError):` in their code to catch both.

Reply:
Okay, makes sense. I have updated the PR.

Reply:
Sorry, I missed this comment when it was requested. @sigmavirus24, I think we may want the `ValueError` here.

All of our other "Invalid*` exceptions inherit from `ValueError` because they're legitimate syntax/value errors. Having it be only `ProxyError` is awkward because this isn't actually a `ConnectionError`, it's raised before anything is sent. If anything this is closer to our `InvalidURL` exception.

Reply:
This explanation for reverting 

```python
class InvalidProxyURL(ProxyError):
```

back to 

```python
class InvalidProxyURL(ProxyError, ValueError):
```

makes sense to me.

Reply:
I have to say that I'm on the fence.

For one thing, I don't think we should ever have been inheriting from `ValueError` except in cases where backwards compatibility was an issue. I suspect we started inheriting from `ValueError` so old code would continue to work without users having to update things. Once we did that the first time, it became a pattern that people copied and pasted. In general, I'll say it seems like a bad pattern to follow.

With that said, this can be resolved by having this exception inherit from `InvalidURL` as you pointed out. This is a special case of an invalid URL and you're right, @nateprewitt, that it isn't a `ConnectionError`.

So if we change this to inherit from `InvalidURL` that seems to be the best way to move forward @nehaljwani

Reply:
I have made the change to make `InvalidProxyURL` inherit from `InvalidURL`